doc: nrf_modem: small updates (#874)
nrfconnect/sdk-nrfxlib: nrf_modem/doc/fault_handling.rst

   The reinitialization of the Modem library must be done outside of the Modem library fault handler.

.. important::

>    If modem traces are enabled, the modem will output keep outputting trace data (a coredump) upon a fault, after having signalled the fault to the application.

Reviewer: `the modem will output keep outputting` - this doesn't sound good.

>    If modem traces are enabled, the modem will output keep outputting trace data (a coredump) upon a fault, after having signalled the fault to the application.

Reviewer 2: Maybe
```suggestion
   If modem traces are enabled, the modem will perform a coredump upon a modem fault. The trace data is output by the modem after signaling the fault to the application.
```

>    If modem traces are enabled, the modem will output keep outputting trace data (a coredump) upon a fault, after having signalled the fault to the application.

Reviewer 3: ```suggestion
   If modem traces are enabled, the modem continues to output trace data (a coredump) in the event of a fault, even after signaling the application of the fault.
```

   To make sure the application is able to retrieve the coredump correctly, the application must not re-initialize the modem until all the outstanding trace data has been processed.
   The :c:func:`nrf_modem_trace_get()` function will return ``-ENODATA`` when all outstanding trace data has been processed by the application.

> When using the Modem library in |NCS|, the Modem library integration layer takes care of synchronizing the modem re-initialization with the tracing operations.

Reviewer 3: ```suggestion
When the Modem library is used in |NCS|, the :ref:`nrf_modem_lib_readme` handles synchronizing modem re-initialization with tracing operations.
```
